components/kyma-environment-broker/cmd/broker/main.go

```diff
@@ -471,6 +474,29 @@ func processOrchestration(state string, op storage.Orchestrations, queue *proces
 	return nil
 }
 
+// processCanceledOrchestration reprocess canceled orchestrations only when some in progress operations exists
+func processCanceledOrchestration(orchestrationsStorage storage.Orchestrations, operationsStorage storage.Operations, queue *process.Queue, log logrus.FieldLogger) error {
```

Comment:
Should we reprocess all cancelled orchestrations every time KEB is started? I know there is a corner case when KEB is restarted/crashes during cancellation, but I'm thinking we should only reprocess the last one, as ne number of cancelled orchestrations could be huge over time.

Reply:
We can implement it to check only the latest operation to decrease the complexity but we need to keep in mind that we might have canceled orchestration with unfinished operations in our storage. I'd prefer to have that mechanism reliable to not have situations like with never-ending provisioning/deprovisioning operations but as you wish :)

Reply:
Added `return nil` after we find the first matching canceled orchestration to reprocess to skip the others